src/select/types.js

```diff
@@ -114,13 +111,11 @@ export type StatefulContainerPropsT = {
 
 export type StatefulSelectPropsT = {
   overrides?: OverridesT,
-  options?: Array<OptionT>,
+  options?: Array<OptionT> | ((query?: *) => Promise<Array<OptionT>>),
```

Comment:
is there any case where query is not a `string`?

Reply:
currently our input provides text as the main query. We could have changed it, if input supported some other objects to be sent or we would have such objects attached to query string

Reply:
yeah it seems like it will always be string, if so you might be able to change the `*` to `string` in your flowtype here.

Reply:
ok